recipe/thirdparty/providers/discord.go

```diff
@@ -16,100 +16,71 @@
 package providers
 
 import (
-	"net/http"
-	"strings"
-
 	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
 	"github.com/supertokens/supertokens-golang/supertokens"
 )
 
 const discordID = "discord"
 
-func Discord(config tpmodels.DiscordConfig) tpmodels.TypeProvider {
-	return tpmodels.TypeProvider{
-		ID: discordID,
-		Get: func(redirectURI, authCodeFromRequest *string, userContext supertokens.UserContext) tpmodels.TypeProviderGetResponse {
-			accessTokenAPIURL := "https://discord.com/api/oauth2/token"
-			accessTokenAPIParams := map[string]string{
-				"client_id":     config.ClientID,
-				"client_secret": config.ClientSecret,
-				"grant_type":    "authorization_code",
-			}
-			if authCodeFromRequest != nil {
-				accessTokenAPIParams["code"] = *authCodeFromRequest
-			}
-			if redirectURI != nil {
-				accessTokenAPIParams["redirect_uri"] = *redirectURI
-			}
+func Discord(input tpmodels.ProviderInput) tpmodels.TypeProvider {
```

Comment:
```suggestion
func Discord(input tpmodels.ProviderInput, requireEmail boolean) tpmodels.TypeProvider {
```

Reply:
Make requireEmail as false by deafult, and if it;s false, we assign fake email to the resulting user.

Reply:
Using addition config instead of param here. Done